### PR TITLE
[GLUTEN-4917][VL] Enable iceberg/delta in new CI

### PR DESCRIPTION
**Pull request by yma11**
## What changes were proposed in this pull request?

Fix broken inheritance of `IcebergScanTransformer` cause by [commit](https://github.com/apache/incubator-gluten/commit/23048fb6a3d1520dd1336371dc67ec6dc5fbd55a), tracked by [issue](https://github.com/apache/incubator-gluten/issues/5250). 
Re-enable iceberg and delta UTs in CI.

## How was this patch tested?

CI

### CODE REVIEW

**Comment by zhouyuan** [comment]
https://github.com/apache/incubator-gluten/pull/5117 -> this is not required, as it's covered in this patch?

File: gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ case class IcebergScanTransformer(
 
   override lazy val fileFormat: ReadFileFormat = GlutenIcebergSourceUtil.getFileFormat(scan)
 
-  override def getSplitInfos: Seq[SplitInfo] = {
+  override def getSplitInfosFromPartitions(partitions: Seq[InputPartition]): Seq[SplitInfo] = {
```
Comment (zhouyuan) [inline gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala:59]: this seems a bug, can you please fire one issue to track for it?

**Comment by yma11** [comment]
> #5117 -> this is not required, as it's covered in this patch?

Yes, I will close it.

File: gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala (hunk 1 above)
Comment (yma11) [reply gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala:59]: [5250](https://github.com/apache/incubator-gluten/issues/5250) created.

Comment (yma11) [inline gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala:59]: @acvictor Can you help review this change?

Comment (acvictor) [reply gluten-iceberg/src/main/scala/org/apache/gluten/execution/IcebergScanTransformer.scala:59]: Looks good to me, thank you!

**Review by acvictor — APPROVED** [review]

File: gluten-delta/src/test/scala/org/apache/gluten/execution/VeloxTPCHDeltaSuite.scala [hunk 2]
```diff
@@ -36,6 +36,7 @@ class VeloxTPCHDeltaSuite extends VeloxTPCHSuite {
 
   override protected def sparkConf: SparkConf = {
     super.sparkConf
+      .set("spark.executor.memory", "4g")
```
[the remark this answers is not in the document]
Comment (yma11) [reply gluten-delta/src/test/scala/org/apache/gluten/execution/VeloxTPCHDeltaSuite.scala:39]: not quite. It leverages existing dataset located in test [resources](https://github.com/apache/incubator-gluten/tree/main/backends-velox/src/test/resources/tpch-data-parquet-velox).

**Comment by yma11** [comment]
@zzcclp can you help take a look at CH CI [failure](https://opencicd.kyligence.com/job/gluten/job/gluten-ci/8517/consoleText)? Thanks.

**Comment by zhouyuan** [comment]
Run Gluten Clickhouse CI

**Review by zhouyuan — APPROVED** [review]

**Pull request merged**